### PR TITLE
RECIRC 198: Only try to load templates if a name is supplied

### PR DESCRIPTION
**Pull request by pauloslund**
# Links
- https://wikia-inc.atlassian.net/browse/RECIRC-198
# Description

We want to check to make sure we're passing in a name for the template to render

### CODE REVIEW

File: extensions/wikia/Recirculation/js/views/rail.js [hunk 1]
```diff
@@ -9,15 +9,13 @@ define('ext.wikia.recirculation.views.rail', [
 ], function ($, w, log, abTest, tracker, utils) {
 
 	var logGroup = 'ext.wikia.recirculation.views.rail',
```
Comment (WikiaTech) [inline extensions/wikia/Recirculation/js/views/rail.js]: MAJOR Remove the declaration of the unused 'logGroup' variable. [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AUnusedVariable)
MAJOR Remove this useless assignment to local variable "logGroup" [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AS1854)

File: extensions/wikia/Recirculation/js/views/impactFooter.js [hunk 2]
```diff
@@ -9,9 +9,7 @@ define('ext.wikia.recirculation.views.impactFooter', [
 
 	var logGroup = 'ext.wikia.recirculation.views.rail',
```
Comment (WikiaTech) [inline extensions/wikia/Recirculation/js/views/impactFooter.js]: MAJOR Remove the declaration of the unused 'logGroup' variable. [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AUnusedVariable)
MAJOR Remove this useless assignment to local variable "logGroup" [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AS1854)

File: extensions/wikia/Recirculation/js/views/impactFooter.js [hunk 3]
```diff
@@ -9,9 +9,7 @@ define('ext.wikia.recirculation.views.impactFooter', [
...
 		imageRatio = 9/16;
-		options = {
-			template: 'impactFooter.mustache'
-		};
+		options = {};
```
Comment (WikiaTech) [inline extensions/wikia/Recirculation/js/views/impactFooter.js]: MAJOR Add the "let", "const" or "var" keyword to this declaration of "options" to make it explicit. [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AS2703)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 13 issues
* MAJOR 7 major
* MINOR 5 minor
* INFO 1 info

Watch the comments in this conversation to review them.

#### 8 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [impactFooter.js#L78](https://github.com/Wikia/app/blob/3d6391c501d45065b7909ae3b7c1c7a121d93e65/extensions/wikia/Recirculation/js/views/impactFooter.js#L78): Remove the unused function parameter "e". [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AUnusedFunctionArgument)
1. MAJOR [impactFooter.js#L85](https://github.com/Wikia/app/blob/3d6391c501d45065b7909ae3b7c1c7a121d93e65/extensions/wikia/Recirculation/js/views/impactFooter.js#L85): Remove the unused function parameter "e". [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AUnusedFunctionArgument)
1. MINOR [utils.js#L39](https://github.com/Wikia/app/blob/3d6391c501d45065b7909ae3b7c1c7a121d93e65/extensions/wikia/Recirculation/js/utils.js#L39): Rename this 'Result' function to match the regular expression ^[a-z][a-zA-Z0-9]*$ [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AS100)
1. MINOR [utils.js#L58](https://github.com/Wikia/app/blob/3d6391c501d45065b7909ae3b7c1c7a121d93e65/extensions/wikia/Recirculation/js/utils.js#L58): Add a semicolon at the end of this statement. [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3ASemicolon)
1. MINOR [rail.js#L32](https://github.com/Wikia/app/blob/3d6391c501d45065b7909ae3b7c1c7a121d93e65/extensions/wikia/Recirculation/js/views/rail.js#L32): Add a semicolon at the end of this statement. [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3ASemicolon)
1. MINOR [rail.js#L48](https://github.com/Wikia/app/blob/3d6391c501d45065b7909ae3b7c1c7a121d93e65/extensions/wikia/Recirculation/js/views/rail.js#L48): Add a semicolon at the end of this statement. [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3ASemicolon)
1. MINOR [rail.js#L51](https://github.com/Wikia/app/blob/3d6391c501d45065b7909ae3b7c1c7a121d93e65/extensions/wikia/Recirculation/js/views/rail.js#L51): Add a semicolon at the end of this statement. [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3ASemicolon)
1. INFO [utils.js#L89](https://github.com/Wikia/app/blob/3d6391c501d45065b7909ae3b7c1c7a121d93e65/extensions/wikia/Recirculation/js/utils.js#L89): Move this trailing comment on the previous empty line. [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3ATrailingComment)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 2 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MINOR [utils.js#L39](https://github.com/Wikia/app/blob/e2e496e63b55c8b321d6bb357f019c7a5e6d7023/extensions/wikia/Recirculation/js/utils.js#L39): Rename this 'Result' function to match the regular expression ^[a-z][a-zA-Z0-9]*$ [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AS100)
1. MINOR [utils.js#L58](https://github.com/Wikia/app/blob/e2e496e63b55c8b321d6bb357f019c7a5e6d7023/extensions/wikia/Recirculation/js/utils.js#L58): Add a semicolon at the end of this statement. [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3ASemicolon)

**Comment by gabrys** [comment]
LGTM

**Pull request merged**